app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,133 @@
 class SendgridApi
-  extend SingleForwardable
+  class Error < StandardError; end
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  TIMEOUT = 2 # seconds
+
+  def initialize
+    @connection = Excon.new(Rails.configuration.sendgrid_api_host,
+      persistent: true,
+      connect_timeout: TIMEOUT)
+  end
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    route = 'spamreports.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def bounced?(email)
-    api { bounces.retrieve(email: email).any? }
+    route = 'bounces.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def remove_from_bounce_list(email)
-    api { bounces.delete(email: email) }
+    route = 'bounces.delete.json'
+    body = { email: email }
+
+    response = post_request(route, body)
+
+    check_email_error(response)
+    true
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def remove_from_spam_list(email)
-    api { spam_reports.delete(email: email) }
-  end
+    route = 'spamreports.delete.json'
+    body = { email: email }
 
-private
+    response = post_request(route, body)
 
-  def api(&_action)
-    yield
+    check_email_error(response)
+    true
   rescue => e
     Rails.logger.error("SendgridApi error: #{e.class} #{e}")
     false
   end
 
-  def spam_reports
-    SendgridToolkit::SpamReports.new
+private
+
+  def credentials
+    if Rails.configuration.sendgrid_api_user.blank? ||
+       Rails.configuration.sendgrid_api_key.blank?
+      fail Error, 'no api credentials specified'
+    end
+
+    {
+      api_user: Rails.configuration.sendgrid_api_user,
+      api_key: Rails.configuration.sendgrid_api_key
+    }
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def get_request(route, query, timeout: TIMEOUT)
+    options = {
+      path: "api/#{route}",
+      expects: [200],
+      headers: {
+        'Accept' => 'application/json',
+        'Accept-Language' => 'en'
+      },
+      query: query.merge(credentials),
+      read_timeout: timeout,
+      write_timeout: timeout
+    }
+
+    response = JSON.parse(@connection.get(options).body)
+
+    check_for_error(response)
+
+    response
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  # rubocop:disable Metrics/MethodLength
+  def post_request(route, body, timeout: TIMEOUT)
+    options = {
+      path: "/api/#{route}",
+      expects: [200],
+      headers: {
+        'Accept' => 'application/json',
+        'Accept-Language' => 'en'
+      },
+      query: URI.encode_www_form(body.merge(credentials)),
```

Comment:
We shouldn't be using query in the case of a POST request. I haven't looked at the sendgrid API design, but I'd be surprised if it doesn't accept `application/json`.